internal/p2p/pex/reactor.go

```diff
@@ -192,11 +197,13 @@ func (r *Reactor) processPexCh() {
 // processPeerUpdates initiates a blocking process where we listen for and handle
 // PeerUpdate messages. When the reactor is stopped, we will catch the signal and
 // close the p2p PeerUpdatesCh gracefully.
-func (r *Reactor) processPeerUpdates() {
+func (r *Reactor) processPeerUpdates(ctx context.Context) {
 	defer r.peerUpdates.Close()
 
 	for {
 		select {
+		case <-ctx.Done():
```

Comment:
Because the peer update channel may still be writable at this point, this may not fully exhaust the race condition. But it should at least reduce how often it gets hit. 🙂